sky/execution.py

```diff
@@ -158,16 +159,32 @@ def _execute(
             cluster_name)
         cluster_exists = existing_handle is not None
 
+    stages = stages if stages is not None else list(Stage)
+
     backend = backend if backend is not None else backends.CloudVmRayBackend()
-    if not isinstance(backend, backends.CloudVmRayBackend
-                     ) and idle_minutes_to_autostop is not None:
+    if isinstance(backend, backends.CloudVmRayBackend):
+        if down and idle_minutes_to_autostop is None:
+            # Use autostop(down) to terminate the cluster after the task is
+            # done. Otherwise, the cluster will be immediately terminated when
+            # the user detach from the execution.
+            idle_minutes_to_autostop = 0
+        if idle_minutes_to_autostop is not None:
+            if idle_minutes_to_autostop == 0:
+                verb = 'torn down' if down else 'stopped'
+                logger.warning(f'{colorama.Fore.LIGHTBLACK_EX}Setting '
+                               'idle_minutes_to_autostop to 1, to avoid '
+                               f'cluster being {verb} during task submission.'
```

Comment:
Why would `-i 0` have this issue of being torn down during task submission? Is that a bug in how the autostop event is implemented?

Reply:
That makes sense. Can we copy this comment into the code where we reset the var to 1?